.gitlab/config/generate_pipelines.py

```diff
@@ -88,14 +127,13 @@ def has_no_status(branch):
 with open(args.template_file, 'r') as fd:
     template_str = fd.read()
 
-    branches = request_as_dict(gl_url + '/repository/branches')
-    branches = map(add_timestamp, branches)
-    branches = filter(is_recent, branches)
-    branches = filter(has_no_status, branches)
-
-    # Select the arg.max most least recent branches
+    branches = request_as_list(gl_url + '/repository/branches')
+    branches = [add_timestamp(branch) for branch in branches]
+    branches = [b for b in branches if is_recent(b)]
     branches = sorted(branches, key=lambda x: x['dt'])
-    branches = itertools.islice(branches, args.max)
+
+    run_n_times = skip_after_n_successes(default_value=False, n=args.max)
+    branches = [b for b in branches if run_n_times(has_no_status, b)]
```

Comment:
It has taken me some time to figure what the heck this is doing, so I think a descriptive comment would make sense here.  The way I'm reading this code it's intent is to "schedule jobs for the next `args.max` (i.e. 2) branches for which github doesn't have an associated status".  Is that correct though?

Reply:
Sorry I should have made a better introduction :angel: . Yep

Reply:
Well, I was suggesting putting a comment in the code so future developers (including me a month from now) get a hint about what's happening.  But I guess that's not strictly necessary.

Reply:
My bad, find the comment added